src/views/UserDetail/UserDetail.js

```diff
@@ -331,6 +334,15 @@ class UserDetail extends React.Component {
     }
   ]
 
+  getAddressesList(addresses, addressTypes) {
```

Comment:
I know this is not a big list but this is still doing n^2 first mapping over `addresses` and then for each address looping through `addressTypes`. 

Could you try to turn `addressTypes` into a hash outside of map with `keyBy`: https://lodash.com/docs/4.17.15#keyBy 

for example:

`const addressTypesById = keyBy(addressTypes, 'id')`

then inside the address map you could just do:

`const addressTypeOption  = addressTypesById[address.addressType]`

Reply:
Thanks, updated according to your comment